test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,250 @@
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil)).must_equal 'authorization=Bearer 0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+    end
+
+    # 4
+    it 'wrong formatted SW_APM_SERVICE_KEY APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
+    end
+
+    # 6
+    it 'correct formatted SW_APM_SERVICE_KEY, successful auth token APM PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['SW_APM_COLLECTOR'] = 'apm.collector.na-01.cloud.solarwinds.com'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      _(endpoint.instance_variable_get(:@token)).must_equal '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      _(endpoint.instance_variable_get(:@service_name)).must_equal 'my_service'
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+    end
+
+    # 7 (for local testing and lambda collector extension)
+    it 'wrong formatted SW_APM_SERVICE_KEY LOCAL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = nil
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://0.0.0.0:4317'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+    end
+
+    # 9
+    it 'correct formatted SW_APM_SERVICE_KEY, successful auth token LOCAL PROTO' do
+      ENV['SW_APM_SERVICE_KEY'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234:my_service'
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://0.0.0.0:4317'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+    end
+
+    # 10 lambda doesn't care about SW_APM_SERVICE_KEY and OTEL_EXPORTER_OTLP_ENDPOINT
+    it 'SW_APM_API_TOKEN invalid inside lambda' do
+      ENV['SW_APM_API_TOKEN'] = nil
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
```

Comment:
`apm-ruby` should be enabled in this case and the one below (inside lambda). By the way, it doesn't seem like the tests are simulating/causing the library to be "in lambda mode", so these two cases probably end up just testing the "outside lambda, agent disabled due to missing/invalid format service key" scenario.

Reply:
Updated.